Source/FullScreenPokemon.ts

```diff
@@ -703,10 +703,11 @@ module FullScreenPokemon {
          * Retrieves the Thing in MapScreener.thingById of the given id.
          * 
          * @param id   An id of a Thing to retrieve.
+         * @param type    The type for the thing being retrieved.
          * @returns The Thing under the given id, if it exists.
          */
-        getThingById(id: string): IThing {
-            return this.MapScreener.thingsById[id];
+        getThingById(id: string, type: string): any {
+            return this.GroupHolder.getGroup(type)[id];
```

Comment:
Can you actually get rid of this `getThingById`? It's not really needed now that GroupHoldr is getting used.